nsis/src/nsis/format/NsisScriptHeader.java

```diff
@@ -17,71 +17,80 @@
 	private int archiveSize;
 	private int compressedHeaderSize;
 	private int flags;
+	private static Structure STRUCTURE;
 
 	public NsisScriptHeader(BinaryReader reader) throws IOException {
-		setMagic(reader.readNextByteArray(NsisConstants.NSIS_MAGIC.length));
+		this.magic = reader.readNextByteArray(NsisConstants.NSIS_MAGIC.length);
 		if (!Arrays.equals(NsisConstants.NSIS_MAGIC, getMagic())) {
-			throw new IOException("not a nsis file.");
+			throw new IOException("Not a nsis file.");
 		}
 
-		setInflatedHeaderSize(reader.readNextInt());
-		setArchiveSize(reader.readNextInt());
-		setCompressedHeaderSize(reader.readNextInt());
-		setFlags(reader.readNextInt());
+		this.inflatedHeaderSize = reader.readNextInt();
+		this.archiveSize = reader.readNextInt();
+		this.compressedHeaderSize = reader.readNextInt();
+		this.flags = reader.readNextInt();
+		checkHeaderCompression(reader);
 	}
 
 	@Override
 	public DataType toDataType() throws DuplicateNameException, IOException {
-		Structure structure = new StructureDataType("script_header", 0);
-		structure.add(STRING, NsisConstants.NSIS_MAGIC.length, "magic", null);
-		structure.add(DWORD, 4, "inf_size", null);
-		structure.add(DWORD, 4, "hdr_size", null);
-		structure.add(DWORD, 4, "cmpr_hdr_size", null);
-		structure.add(DWORD, 4, "flags", null);
-		return structure;
+		if(STRUCTURE == null) {
+			STRUCTURE = new StructureDataType("script_header", 0);
+			STRUCTURE.add(STRING, NsisConstants.NSIS_MAGIC.length, "magic", null);
+			STRUCTURE.add(DWORD, DWORD.getLength(), "inf_size", null);
+			STRUCTURE.add(DWORD, DWORD.getLength(), "hdr_size", null);
+			STRUCTURE.add(DWORD, DWORD.getLength(), "cmpr_hdr_size", null);
+			STRUCTURE.add(DWORD, DWORD.getLength(), "flags", null);
```

Comment:
Consider using static initialization for STRUCTURE: https://docs.oracle.com/javase/tutorial/java/javaOO/initial.html. In this case you will be able to declare it final and leave the name in uppercase.

Reply:
Changed. TIL it is possible to init static objects like that, thanks!